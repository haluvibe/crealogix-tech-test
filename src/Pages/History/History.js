import React from 'react';
import { Link } from 'react-router-dom';

import CounterProvider from '../../Provider/CounterProvider';
import List from '../../Components/List/List';

const History = () => (
  <>
    <CounterProvider
    	render={({ history }) => (
    		<List
    			history={history}
    		/>
    	)}
    />
    <Link to="/">Counter</Link>
  </>
);

export default History;
