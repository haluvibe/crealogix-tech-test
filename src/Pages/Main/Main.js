import React from "react";
import { Link } from "react-router-dom";

import CounterProvider from "../../Provider/CounterProvider";
import Counter from "../../Components/Counter/Counter";

const Main = () => (
  <>
    <CounterProvider
      render={({ count, handleIncrementCounter, handleDecrementCounter }) => (
        <Counter
          count={count}
          handleIncrementCounter={handleIncrementCounter}
          handleDecrementCounter={handleDecrementCounter}
        />
      )}
    />
    <Link to="/history">History</Link>
  </>
);

export default Main;
