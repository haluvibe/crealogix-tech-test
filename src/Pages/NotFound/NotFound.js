import React from 'react';
import PropTypes from 'prop-types';

const NotFound = () => {
	return (
    <>NotFound Page</>
	);
};

NotFound.propTypes = {};

export default NotFound;
