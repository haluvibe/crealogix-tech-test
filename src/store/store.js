// src/js/store/store.js
import { createStore, applyMiddleware } from 'redux';
import rootReducer from './rootReducer';
import logger from 'redux-logger';

const middlewares = [logger];
const createStoreWithMiddleware = applyMiddleware(...middlewares)(createStore);
const store = createStoreWithMiddleware(rootReducer);
export default store;
