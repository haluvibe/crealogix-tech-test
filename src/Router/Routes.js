// Root.js
import React from "react";
import { BrowserRouter, Route, Switch } from "react-router-dom";

import Main from "../Pages/Main/Main";
import History from "../Pages/History/History";
import NotFound from "../Pages/NotFound/NotFound";

export const ROOT_PATH = "/";
export const HISTORY_PATH = "/history";

const Routes = () => (
  <Switch>
    <Route exact path={ROOT_PATH} component={Main} />
    <Route exact path={HISTORY_PATH} component={History} />
    <Route component={NotFound} />
  </Switch>
);

export default Routes;
