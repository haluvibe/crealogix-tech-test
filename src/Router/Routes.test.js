import React from "react";
import { Provider } from "react-redux";
import { mount, shallow } from "enzyme";
import { MemoryRouter } from "react-router-dom";

import store from "../store/store";
import Routes, { ROOT_PATH, HISTORY_PATH } from "./Routes";
import Main from "../Pages/Main/Main";
import History from "../Pages/History/History";
import NotFound from "../Pages/NotFound/NotFound";

const INVALID_PATH = "/invalid";

const setUp = url => {
  const wrapper = mount(
    <Provider store={store}>
      <MemoryRouter initialEntries={[url]}>
        <Routes />
      </MemoryRouter>
    </Provider>
  );
  return wrapper;
};

describe("Routes", () => {
  let wrapper;
  it("should render correctly with no props", () => {
    wrapper = shallow(<Routes />);

    expect(wrapper).toMatchSnapshot();
  });

  test("invalid path should load the Not Found Page", () => {
    wrapper = setUp(INVALID_PATH);
    expect(wrapper.find(Main)).toHaveLength(0);
    expect(wrapper.find(NotFound)).toHaveLength(1);
  });

  test("root path should load only the Main Page Component", () => {
    wrapper = setUp(ROOT_PATH);
    expect(wrapper.find(Main)).toHaveLength(1);
    expect(wrapper.find(NotFound)).toHaveLength(0);
  });

  test("history path should load only the History Page Component", () => {
    wrapper = setUp(HISTORY_PATH);
    expect(wrapper.find(History)).toHaveLength(1);
    expect(wrapper.find(NotFound)).toHaveLength(0);
  });
});
