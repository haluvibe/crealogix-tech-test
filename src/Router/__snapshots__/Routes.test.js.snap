// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes should render correctly with no props 1`] = `
<Switch>
  <Route
    component={[Function]}
    exact={true}
    path="/"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/history"
  />
  <Route
    component={[Function]}
  />
</Switch>
`;
